From: Rod Whitby <rod@whitby.id.au>
Subject: Corrections to DSM-G600 RevA BE support

The vendor firmware loads the kernel at 0x00600000, so make Apex do
the same.  Move the apex environment into the apex partition (which is
just before the reserved RedBoot partitions).  Also enable the memtest
command.

Signed-off-by: Rod Whitby <rod@whitby.id.au>
--

Index: apex/src/mach-ixp42x/slugos-dsmg600-armeb_config
===================================================================
--- apex.orig/src/mach-ixp42x/slugos-dsmg600-armeb_config	2008-05-28 10:53:47.000000000 +0930
+++ apex/src/mach-ixp42x/slugos-dsmg600-armeb_config	2008-05-28 10:53:53.000000000 +0930
@@ -17,7 +17,7 @@
 #
 # General Setup
 #
-CONFIG_TARGET_DESCRIPTION="SlugOS DMSG600 (bigendian)"
+CONFIG_TARGET_DESCRIPTION="SlugOS DSM-G600/BE"
 CONFIG_CROSS_COMPILE=""
 # CONFIG_AEABI is not set
 CONFIG_CC_OPTIMIZE_FOR_SIZE=y
@@ -38,7 +38,7 @@
 CONFIG_ATAG_PHYS=0x00000100
 CONFIG_ARCH_NUMBER=964
 CONFIG_APEX_VMA=0x00200000
-CONFIG_KERNEL_LMA=0x00400000
+CONFIG_KERNEL_LMA=0x00600000
 # CONFIG_USE_RAMDISK is not set
 CONFIG_SDRAM_BANK0=y
 # CONFIG_SDRAM_BANK1 is not set
@@ -89,7 +89,7 @@
 CONFIG_CMD_XRECEIVE=y
 CONFIG_CMD_MEMLIMIT=y
 CONFIG_CMD_MEMSCAN=y
-# CONFIG_CMD_MEMTEST is not set
+CONFIG_CMD_MEMTEST=y
 
 #
 # Generic Drivers
@@ -114,7 +114,7 @@
 CONFIG_ENV=y
 CONFIG_ENV_LINK=y
 CONFIG_ENV_MUTABLE=y
-CONFIG_ENV_REGION="nor:0xee0000+15k"
+CONFIG_ENV_REGION="nor:0xfbc000+15k"
 # CONFIG_ENV_SAVEATONCE is not set
 CONFIG_ENV_CHECK_LEN=1024
 # CONFIG_VARIATIONS is not set
